source/dbref.rst

```diff
@@ -0,0 +1,349 @@
+
+- ``$ref``: required string field. Contains the name of the collection where
+  the referenced document resides. This MUST be the first field in the DBRef.
+- ``$id``: required field. Contians the value of the ``_id`` field of the
+  referenced document. This MUST be the second field in the DBRef.
+- ``$db``: optional string field. Contains the name of the database where the
+  referenced document resides. If specified, this MUST be the third field in the
+  DBRef. If omitted, the referenced document is assumed to reside in the same
+  database as the DBRef.
+- Extra, optional fields may follow after ``$id`` or ``$db`` (if specified).
+  There are no inherent restrictions on extra field names; however, older server
+  versions may impose their own restrictions (e.g. no dots or dollars).
+
+DBRefs have no relation to the deprecated DBPointer BSON type (i.e. type 0x0C).
+
+
+Examples of Valid DBRefs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following examples are all valid DBRefs:
+
+.. code:: typescript
+
+  // Basic DBRef with only $ref and $id fields
+  { "$ref": "coll0", "$id": { "$oid": "60a6fe9a54f4180c86309efa" } }
+
+  // DBRef $id is not necessarily an ObjectId
+  { "$ref": "coll0", "$id": 1 }
+
+  // DBRef with optional $db field
+  { "$ref": "coll0", "$id": 1, "$db": "db0" }
+
+  // DBRef with extra, optional fields (with or without $db)
+  { "$ref": "coll0", "$id": 1, "$db": "db0", "foo": "bar" }
+  { "$ref": "coll0", "$id": 1, "foo": true }
+
+  // Extra field names have no inherent restrictions
+  { "$ref": "coll0", "$id": 1, "$foo": "bar" }
+  { "$ref": "coll0", "$id": 1, "foo.bar": 0 }
+
+Examples of Invalid DBRefs
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following examples are all invalid DBRefs:
+
+.. code:: typescript
+
+  // Required fields are omitted
+  { "$ref": "coll0" }
+  { "$id": { "$oid": "60a6fe9a54f4180c86309efa" } }
+
+  // Invalid types for $ref or $db
+  { "$ref": true, "$id": 1 }
+  { "$ref": "coll0", "$id": 1, "$db": 1 }
+
+  // Fields are out of order
+  { "$id": 1, "$ref": "coll0" }
+
+
+Implementing a DBRef Model
+--------------------------
+
+Drivers MAY provide a model class for encoding and/or decoding DBRef documents.
+For those drivers that do, this section defines expected behavior of that class.
+This section does not prohibit drivers from implementing additional
+functionality, provided it does not conflict with any of these guidelines.
+
+
+Constructing a DBRef model
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Drivers MAY provide an API for constructing a DBRef model directly from its
+constituent parts. If so:
+
+- Drivers MUST solicit a string value for ``$ref``.
+
+- Drivers MUST solicit an arbitary value for ``$id``. Drivers SHOULD NOT enforce
+  any restrictions on this value; however, this may be necessary if the driver
+  is unable to differentiate between certain BSON types (e.g. ``null``,
+  ``undefined``) and the parameter being unspecified.
+
+- Drivers SHOULD solicit an optional string value for ``$db``.
```

Comment:
why is this a SHOULD rather than a MUST?

Reply:
I was keeping this flexible in case some implementations didn't support this and didn't care to add it. I'm not sure how common cross-DB references (or even cross-DB applications) are in the wild, but I figured even if the model class didn't support this an application could always construct the embedded document on its own.

In my mind, `$ref` and `$id` are the essentials (MUST), `$db` was relevant but still optional (SHOULD), and the extra fields are nice-to-haves. This also only pertains to constructing a DBRef model from individual parameters. The decoding code paths MUST support all of these to ensure DBRefs can be round-tripped without data loss (excluding possible field re-ordering).

Reply:
that makes sense. thanks for the explanation!